src/EventListener/CheckoutCartAmazonPayExpressEventListener.php

```diff
@@ -0,0 +1,173 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PayonePayment\EventListener;
+
+use Exception;
+use PayonePayment\Components\ConfigReader\ConfigReaderInterface;
+use PayonePayment\Components\GenericExpressCheckout\CartExtensionService;
+use PayonePayment\Components\GenericExpressCheckout\Struct\CreateExpressCheckoutSessionStruct;
+use PayonePayment\Components\PaymentFilter\DefaultPaymentFilterService;
+use PayonePayment\Components\PaymentFilter\PaymentFilterContext;
+use PayonePayment\PaymentHandler\PayoneAmazonPayExpressPaymentHandler;
+use PayonePayment\Payone\Client\PayoneClientInterface;
+use PayonePayment\Payone\RequestParameter\RequestParameterFactory;
+use Psr\Log\LoggerInterface;
+use RuntimeException;
+use Shopware\Core\Checkout\Cart\Cart;
+use Shopware\Core\Checkout\Cart\SalesChannel\CartService;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepository;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\Struct\ArrayStruct;
+use Shopware\Core\System\Language\LanguageEntity;
+use Shopware\Core\System\SalesChannel\SalesChannelContext;
+use Shopware\Storefront\Page\Checkout\Cart\CheckoutCartPageLoadedEvent;
+use Shopware\Storefront\Page\Checkout\Offcanvas\OffcanvasCartPageLoadedEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class CheckoutCartAmazonPayExpressEventListener implements EventSubscriberInterface
+{
+    public function __construct(
+        private readonly RequestParameterFactory $requestParameterFactory,
+        private readonly PayoneClientInterface $payoneClient,
+        private readonly LoggerInterface $logger,
+        private readonly ConfigReaderInterface $configReader,
+        private readonly EntityRepository $languageRepository,
+        private readonly DefaultPaymentFilterService $paymentFilterService,
+        private readonly CartExtensionService $cartExtensionService,
+        private readonly CartService $cartService
+    ) {
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            CheckoutCartPageLoadedEvent::class => 'onCartLoaded',
+            OffcanvasCartPageLoadedEvent::class => 'onCartLoaded',
+        ];
+    }
+
+    public function onCartLoaded(CheckoutCartPageLoadedEvent|OffcanvasCartPageLoadedEvent $event): void
+    {
+        /*$paymentMethod = $event->getSalesChannelContext()->getSalesChannel()->getPaymentMethods()->get(PayoneAmazonPay::UUID);
+        if (!$paymentMethod instanceof PaymentMethodEntity || !$paymentMethod->getActive()) {
+            // payment method is not assigned to sales-channel of is disabled.
+            return;
+        }*/
```

Comment:
thank you for the notice. 
I missed to add the functionality for this. 

sadly the payment methods are not loaded within the sales-channel-context, and we would have to load them separately. 
Because we only want to check if the payment method is activated, without loading it, I used a direct SQL Query. 
Got this thought from the PayPal Extension.